test/e2e/e2e.go

```diff
@@ -742,6 +727,133 @@ var _ = Describe("validation", func() {
 			}, timeout, interval).Should(BeTrue())
 		})
 	})
+	Context("Validate AddressPool Webhook", func() {
+		BeforeEach(func() {
+			By("Checking if validation webhook is enabled")
+			deploy, err := testclient.Client.Deployments(OperatorNameSpace).Get(context.Background(), consts.MetalLBOperatorDeploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			isValidationWebhookEnabled := false
+			for _, container := range deploy.Spec.Template.Spec.Containers {
+				if container.Name == "manager" {
+					for _, env := range container.Env {
+						if env.Name == "ENABLE_OPERATOR_WEBHOOK" {
+							if env.Value == "true" {
+								isValidationWebhookEnabled = true
+							}
+						}
+					}
+				}
+			}
+
+			if !isValidationWebhookEnabled {
+				Skip("AddressPool webhook is disabled")
+			}
+
+			By("Checking if validation webhook is running")
+			// Can't just check the ValidatingWebhookConfiguration name as it's changing between different deployment methods.
+			// Need to check the webhook name in the webhooks definition of the ValidatingWebhookConfiguration.
+			validateCfgList := &admv1.ValidatingWebhookConfigurationList{}
+			err = testclient.Client.List(context.TODO(), validateCfgList, &goclient.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			isValidationWebhookRunning := false
+			for _, validateCfg := range validateCfgList.Items {
+				for _, webhook := range validateCfg.Webhooks {
+					if webhook.Name == consts.AddressPoolValidationWebhookName {
+						isValidationWebhookRunning = true
+					}
+				}
+			}
+			Expect(isValidationWebhookRunning).To(BeTrue(), "AddressPool webhook is not running")
+		})
+		It("Should recognize duplicate definition of AddressPools", func() {
+			By("Creating first AddressPool resource")
+			autoAssign := false
+			firstAddressPool := &metallbv1alpha1.AddressPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-addresspool",
+					Namespace: OperatorNameSpace,
+				},
+				Spec: metallbv1alpha1.AddressPoolSpec{
+					Protocol: "layer2",
+					Addresses: []string{
+						"1.1.1.1-1.1.1.100",
+					},
+					AutoAssign: &autoAssign,
+				},
+			}
+			err := testclient.Client.Create(context.Background(), firstAddressPool)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating second AddressPool resource with already defined name")
+			secondAdressPool := &metallbv1alpha1.AddressPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-addresspool",
+					Namespace: OperatorNameSpace,
+				},
+				Spec: metallbv1alpha1.AddressPoolSpec{
+					Protocol: "layer2",
+					Addresses: []string{
+						"1.1.1.101-1.1.1.200",
+					},
+					AutoAssign: &autoAssign,
+				},
+			}
+			err = testclient.Client.Create(context.Background(), secondAdressPool)
+			Expect(err).ToNot(BeNil())
+			if !strings.Contains(fmt.Sprint(err), "duplicate definition of pool") {
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("Creating second AddressPool resource with overlaping addresses")
```

Comment:
nit: typo overlaping -> overlapping
another thing, I think we should name this Creating .. defined by address range or something like that, so we have 2 Bys with the same description